packages/core/src/Stores/client-store.js

```diff
@@ -1272,7 +1272,7 @@ export default class ClientStore extends BaseStore {
         this.accounts[this.loginid].is_virtual = +response.authorize.is_virtual;
         this.accounts[this.loginid].session_start = parseInt(moment().utc().valueOf() / 1000);
         this.accounts[this.loginid].landing_company_shortcode = response.authorize.landing_company_name;
-        this.accounts[this.loginid].country = response.country;
+        this.accounts[this.loginid].country = response.authorize.country;
```

Comment:
In some places of our code base, e.g. in cashier, when the user goes to deposit page (using `https://localhost.binary.sx/cashier/deposit` url directly) we should show him PA tab. To show this tab we first need to check the PA list. To check the list we need to send a request with current residence and currency of the client. 

Earlier we used `await ws.wait('get_settings')` method which should request the residence of the client and pass it to the request function, but the residence value was updated to late and our residence was an empty string (""). 

Since we call an authorize request **one of the first!** every time when the user directly goes to any URL, we can update the residence of the client much faster then we check the "get_settings" request. The same thing we do with the currency of the client:
```
get currency() {
        if (this.selected_currency.length) {
            return this.selected_currency;
        **} else if (this.is_logged_in) {**
            **return this.accounts[this.loginid].currency**;
        }

        return this.default_currency;
    }
```
So that solves the issues when you need to make a request with currency and residence parameters on first loading (doesn't matter which url you try to reach).